pandas/core/arrays/interval.py

```diff
@@ -557,15 +561,8 @@ def __setitem__(self, key, value):
 
         key = check_array_indexer(self, key)
 
-        # Need to ensure that left and right are updated atomically, so we're
-        # forced to copy, update the copy, and swap in the new values.
-        left = self.left.copy(deep=True)
-        left._values[key] = value_left
-        self._left = left
```

Comment:
Do you know the reason this was originally done like this?

Reply:
The comment on 560-561 suggests its to ensure setting it atomic

Reply:
It turns out this was indirectly the source of the failures.  Or more specifically, this incorrect behavior is preventing the failures in master.

When we do `IntervalArray.from_breaks` we end up passing `left = orig[:-1]` and `right = orig[1:]`, so then when we later do `arr[1] = None` without this copy, that sets both left[1] and right[1], i.e. `orig[1]` and `orig[2]`, at which point the IntervalArray is corrupted.

Reply:
Ah, nice catch. Can you add an explicit test for it? (eg asserting that left and right are not viewing the same data in from_breaks) That will be more explicit if it fails again in the future than the seemingly unrelated arrow roundtrip failure ..

Reply:
IIRC the test this un-xfails in tests.extension.test_interval was about this problem.  Do you have something more specific in mind?

Reply:
The test that is now activated is about setitem preserving view (and the copy of the full data in setitem of course prevented this). But I meant the view/copy behaviour upon *construction*, not setitem. Of course, no longer copying on setitem revealed that we were creating views upon construction, but I still think it is valuable to have a specific test about the construction part ("eg asserting that left and right are not viewing the same data in from_breaks")